h1_syscalls/src/lib.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![crate_name = "h1_syscalls"]
+#![crate_type = "rlib"]
+#![no_std]
+#![feature(asm, core_intrinsics, const_fn, naked_functions)]
```

Comment:
Why do we need `naked_functions`?

Reply:
We actually didn't need any of those features. Sorry for bad copy&paste, I'm still figuring out what's important ...